xbmc/cores/VideoPlayer/DVDInputStreams/DVDFactoryInputStream.cpp

```diff
@@ -175,6 +175,9 @@ CDVDInputStream* CDVDFactoryInputStream::CreateInputStream(IVideoPlayer* pPlayer
 
     if (finalFileitem.GetMimeType() == "application/vnd.apple.mpegurl")
       return new CDVDInputStreamFFmpeg(finalFileitem);
+
+    if (URIUtils::IsProtocol(finalFileitem.GetPath(), "udp"))
+      return CreateInputStream(pPlayer, finalFileitem, scanforextaudio);
```

Comment:
returning CDVDInputStreamFFmpeg seems a more robust solution to me. if udp was not handled before this line, you would end up in a stack overflow